library/src/blas3/Tensile/gemm_strided_batched.cpp

```diff
@@ -195,20 +182,92 @@ namespace
             }
         }
 
-        rocblas_status validArgs = validateArgs(handle, trans_a, trans_b,
-                                        m, n, k, alpha,
-                                        A, ld_a, stride_a,
-                                        B, ld_b, stride_b, beta,
-                                        C, ld_c, stride_c, b_c);
+#ifdef USE_TENSILE_HOST
+
+        T alpha_h;
+        T beta_h;
+        if(rocblas_pointer_mode_host == handle->pointer_mode)
+        {
+            alpha_h = *alpha;
+            beta_h  = *beta;
+        }
+        else
+        {
+            hipMemcpy(&alpha_h, alpha, sizeof(T), hipMemcpyDeviceToHost);
+            hipMemcpy(&beta_h, beta, sizeof(T), hipMemcpyDeviceToHost);
+        }
+
+        RocblasContractionProblem<T> problem(ContractionProblemType::GEMMStridedBatch,
+                                             trans_a,
+                                             trans_b,
+                                             m,
+                                             n,
+                                             k,
+                                             alpha_h,
+                                             A,
+                                             ld_a,
+                                             stride_a,
+                                             B,
+                                             ld_b,
+                                             stride_b,
+                                             beta_h,
+                                             C,
+                                             ld_c,
+                                             stride_c,
+                                             b_c);
+
+        return handle->host->runContractionProblem(problem);
+
+#else
+        rocblas_status validArgs = validateArgs(handle,
+                                                trans_a,
```

Comment:
This argument validation needs to be moved out of the `#ifdef USE_TENSILE_HOST` and executed for both the old and new client code.